tests/test_browser.go

```diff
@@ -123,11 +124,24 @@ func newTestBrowser(tb testing.TB, opts ...interface{}) *testBrowser {
 	}
 
 	// launch the browser
+	initEnv := &k6common.InitEnvironment{
+		Logger:   state.Logger,
+		Registry: registry,
+	}
+	// NewBrowserType must be run from the init context, but since the recent
+	// validation introduced in k6/js/modulestest, InitEnvField and StateField
+	// must not be both set at the same time. So we nil StateField, create
+	// the BrowserType, and revert it before running Launch().
+	// See https://github.com/grafana/k6/pull/2602
+	vu.InitEnvField = initEnv
+	vu.StateField = nil
```

Comment:
I have a funny feeling about this fix :flushed: Should we better work on a proper fix in the next cycle? 🤔

Reply:
Yes, this is temporary and we should find a better way, hopefully with @mstoykov's help.

Reply:
🤞

Reply:
Something like https://github.com/grafana/xk6-browser/pull/525 will is how the API is more appropriate to be used.